### PR TITLE
Silence extra diagnostics in the testsuite

### PR DESCRIPTION
**Pull request by tamiko**

### CODE REVIEW

File: tests/tests.h [hunk 1]
```diff
@@ -32,6 +32,11 @@
 #include <sstream>
 #include <iomanip>
 
+// silence extra diagnostics in the testsuite
+#ifdef DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+#endif
```
Comment (bangerth) [inline tests/tests.h]: Since you `#include` config.h, shouldn't it always be `#define`d (possibly to the empty string)?

Also, this disables *all* diagnostics for *all* tests. Why is this necessary? I would think that there is value in these warnings, even if currently we don't do anything with them.

Comment (tamiko) [inline tests/tests.h]: Nowadays, the testsuite can be configured independently. While it is clear that changes in the library and the testsuite are somewhat tightly coupled, I think it has some value to not break every test when configured against a deal.II library that doesn't provide ```DEAL_II_DISABLE_EXTRA_DIAGNOSTICS```.

Comment (bangerth) [inline tests/tests.h]: Shrug. Why would you run the latest testsuite against an old deal.II installation that doesn't export this `#define`? You'd get dozens of failures for other reasons as well.

But you didn't answer why you want to do this in the first place?

Comment (tamiko) [inline tests/tests.h]: Because we hit 3736 warnings within the testsuite resulting in 3MB of additional output that has to be stored and processed by the cdash instance.

Please also note that ```DEAL_II_DISABLE_EXTRA_DIAGNOSTICS``` does <i>not</i> disable <i>all</i> diagnostics, but only - in essence - ```-Wextra``` (the category, not the compiler flag), ```-Wunused-*```, and two more.

Comment (bangerth) [inline tests/tests.h]: OK, fair enough.
